command/executor_plugin.go

```diff
@@ -69,3 +84,75 @@ func (e *ExecutorPluginCommand) Run(args []string) int {
 	})
 	return 0
 }
+
+type circbufWriter struct {
+	buf     *circbuf.Buffer
+	err     error
+	bufLock sync.Mutex
+	wr      io.Writer
+	flushCh chan struct{}
+}
+
+func newCircbufWriter(w io.Writer) *circbufWriter {
+	buf, _ := circbuf.NewBuffer(circleBufferSize)
+	c := &circbufWriter{
+		buf:     buf,
+		wr:      w,
+		flushCh: make(chan struct{}),
+	}
+	go c.flushLoop()
+	return c
+}
+
+func (c *circbufWriter) Write(p []byte) (nn int, err error) {
+	if c.err != nil {
+		return nn, c.err
+	}
+	c.bufLock.Lock()
+	nn, err = c.buf.Write(p)
+	c.bufLock.Unlock()
+
+	select {
+	case c.flushCh <- struct{}{}:
+	default:
+	}
+	return nn, err
+}
+
+func (c *circbufWriter) Close() error {
+	var err error
+	if wc, ok := c.wr.(io.WriteCloser); ok {
+		err = wc.Close()
+	}
+
+	close(c.flushCh)
```

Comment:
+1 to notnoop's comment but also protect against double closes here as it's common in tests to defer a Close in case of test failure but also explicitly call Close at a specific point in the test.

```suggestion
	select {
	case <-c.flushCh:
	default:
		close(c.flushCh)
	}
```